src/MongoDB.Analyzer/Core/ReferencesProvider.cs

```diff
@@ -98,6 +102,11 @@ public static ReferencesContainer GetReferences(IEnumerable<MetadataReference> m
             TryAddingAssembly(dependency);
         }
 
+        if (analysisType == AnalysisType.Poco)
+        {
+            CompilePocoAnalysisCode(resultReferences);
```

Comment:
So is this static code compilation is done for each analysis? Or is this memory assembly cached somewhere?

Reply:
I modified the code as follows: 
1. the code looks for the raw assembly bytes based on the data stored in the application domain, 
2. If found, then all we have to do is add a reference to it
3. If not found, we have to create the compilation, cache it in the Application Domain(i.e. currentAppDomain.SetData) and create the reference to it 

So essentially, after the modifications, the static code compilation is done only once and the memory assembly raw bytes are cached in the current ApplicationDomain